# Загрузка сообщений из Телеграмм-канала 

Скрипт для выгрузки сообщений канала был взят c GitHub: [tg_downloader](https://github.com/bezgubov/tg_downloader)

С его помощью выгружены все сообщения канала [MarketTwits](https://t.me/markettwits) в файл channel_messages.json.  
Ранее был выбран период для обучения 4 года, загрузку сообщений канала произведем за выбранный период.

In [1]:
# Выгрузка происходит продолжительное время, запустить в консоли
# ! python tg_downloader.py https://t.me/markettwits --date-from 2022-08-01 --date-to 2026-07-31 --out-dir ./data

In [2]:
import pandas as pd

## Загрузка и первичный обзор данных

In [3]:
df_msgs = pd.read_json('./data/markettwits_posts_2022-08-01_2026-07-31/posts_markettwits_2022-08-01_2026-07-31.json')
df_msgs.info()

<class 'pandas.DataFrame'>
RangeIndex: 172695 entries, 0 to 172694
Data columns (total 8 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   id         172695 non-null  int64  
 1   date_utc   172695 non-null  str    
 2   message    172695 non-null  str    
 3   views      172694 non-null  float64
 4   forwards   172694 non-null  float64
 5   replies    0 non-null       float64
 6   has_media  172695 non-null  bool   
 7   link       172695 non-null  str    
dtypes: bool(1), float64(3), int64(1), str(3)
memory usage: 9.4 MB


In [4]:
df_msgs.head(10)

,id,date_utc,message,views,forwards,replies,has_media,link
0,379587,2026-07-31T21:01:39+00:00,❗️⚠️🇮🇷#иран #геополитика \nСША И ИЗРАИЛЬ ГОТОВ...,71667.0,384.0,NaN,True,https://t.me/markettwits/379587
1,379586,2026-07-31T19:06:20+00:00,"🇮🇷#иран #today #трамп \nТрамп заявил, что Уитк...",64653.0,83.0,NaN,True,https://t.me/markettwits/379586
2,379585,2026-07-31T18:25:15+00:00,🚫✴️🇷🇺#майнинг #россия #крипто\nПравительство Р...,64089.0,195.0,NaN,False,https://t.me/markettwits/379585
3,379584,2026-07-31T18:22:45+00:00,"🇺🇸🇷🇺#санкции #россия \nЗаконопроект об ""адских...",65957.0,93.0,NaN,False,https://t.me/markettwits/379584
4,379583,2026-07-31T18:04:08+00:00,❗️🇯🇵 #fx #япония #интервенции\nСША и Япония пр...,60450.0,71.0,NaN,False,https://t.me/markettwits/379583
5,379582,2026-07-31T17:58:52+00:00,"⚠️🇷🇺#акции #россия \nПо состоянию на 21 июля, ...",67108.0,399.0,NaN,True,https://t.me/markettwits/379582
6,379581,2026-07-31T16:24:01+00:00,🐲 Китайская народная мудрость гласит: смотри И...,57854.0,43.0,NaN,True,https://t.me/markettwits/379581
7,379580,2026-07-31T16:04:45+00:00,⚠️✴️#BTC #крипто #презентация \nМассивной капи...,54854.0,253.0,NaN,True,https://t.me/markettwits/379580
8,379579,2026-07-31T15:48:44+00:00,⚠️🇮🇷#иран #геополитика #трамп \nТрамп: Скоро у...,54301.0,78.0,NaN,False,https://t.me/markettwits/379579
9,379578,2026-07-31T15:25:12+00:00,✴️#Tether #крипто #отчетность \nTETHER - ПРИБЫ...,55999.0,127.0,NaN,False,https://t.me/markettwits/379578


In [5]:
print(df_msgs['date_utc'].min())
print(df_msgs['date_utc'].max())

2022-08-01T04:19:42+00:00
2026-07-31T21:01:39+00:00


In [6]:
df_msgs.describe(include='all')

,id,date_utc,message,views,forwards,replies,has_media,link
count,172695.000000,172695,172695,172694.000000,172694.000000,0.0,172695,172695
unique,NaN,172028,165669,NaN,NaN,NaN,2,172695
top,NaN,2024-12-30T15:37:12+00:00,🌎#markets,NaN,NaN,NaN,False,https://t.me/markettwits/379587
freq,NaN,14,996,NaN,NaN,NaN,121986,1
mean,291395.388494,NaN,NaN,45068.069209,114.478731,NaN,NaN,NaN
std,50981.529208,NaN,NaN,11234.182665,177.914186,NaN,NaN,NaN
min,203002.000000,NaN,NaN,6902.000000,0.000000,NaN,NaN,NaN
25%,247269.500000,NaN,NaN,36082.000000,36.000000,NaN,NaN,NaN
50%,291500.000000,NaN,NaN,42709.000000,66.000000,NaN,NaN,NaN
75%,335513.500000,NaN,NaN,52362.000000,127.000000,NaN,NaN,NaN


Датасет содержит следующие столбцы:
- **id** - идентификатор сообщения в Телеграме (*подлежит удалению*);
- **date_utc** - время публикации сообщения;
- **message** - текст сообщения;
- **views** - количество просмотров (изменяется во времени, актуальное значение можно получить только в моменте, значение можно использовать при построении системы, работающей в режиме реального времени, для определения степени распространения информации и использовать для улучшения работы модели в последующих итерациях разработки, *подлежит удалению*);
- **forwards** - количество ретрансляций сообщения (комментарий тот же, что и для views, *подлежит удалению*);
- **replies** - все значения пустые, для канала не актуально (*подлежит удалению*);
- **has_media** - наличие прикрепленной мультимедийной информации (*подлежит удалению*, мультимедиа на данной итерации не обрабатывается);
- **link** - ссылка на сообщение (*подлежит удалению*).

Выгруженные данные содержат сообщения канала с 7:19 1 августа 2022 г. по 00:01 1 августа 2026 г., 172695 записей сообщений.
Произведем первичную очистку датасета от отмеченных столбцов, а также от записей без текста сообщений.

In [7]:
drop_columns = ['id', 'views', 'forwards', 'replies', 'has_media', 'link']
df_msgs_clr = df_msgs.drop(columns=drop_columns)
df_msgs_clr.info()

<class 'pandas.DataFrame'>
RangeIndex: 172695 entries, 0 to 172694
Data columns (total 2 columns):
 #   Column    Non-Null Count   Dtype
---  ------    --------------   -----
 0   date_utc  172695 non-null  str  
 1   message   172695 non-null  str  
dtypes: str(2)
memory usage: 2.6 MB


In [8]:
df_msgs_clr = df_msgs_clr[df_msgs_clr['message'] != '']
df_msgs_clr

,date_utc,message
0,2026-07-31T21:01:39+00:00,❗️⚠️🇮🇷#иран #геополитика \nСША И ИЗРАИЛЬ ГОТОВ...
1,2026-07-31T19:06:20+00:00,"🇮🇷#иран #today #трамп \nТрамп заявил, что Уитк..."
2,2026-07-31T18:25:15+00:00,🚫✴️🇷🇺#майнинг #россия #крипто\nПравительство Р...
3,2026-07-31T18:22:45+00:00,"🇺🇸🇷🇺#санкции #россия \nЗаконопроект об ""адских..."
4,2026-07-31T18:04:08+00:00,❗️🇯🇵 #fx #япония #интервенции\nСША и Япония пр...
...,...,...
172690,2022-08-01T04:33:34+00:00,⚠️🇷🇸#сербия #косово #геополитика #нато \nМисси...
172691,2022-08-01T04:25:18+00:00,🌎#ты
172692,2022-08-01T04:23:28+00:00,🌎#FX
172693,2022-08-01T04:22:52+00:00,🌎#рынки


Предварительно очищенный датасет сообщений канала содержит 171856 сообщений. Для дальнейшей обработки и анализа данных выгружаем датасет в файл.

In [9]:
df_msgs_clr.to_csv('./data/tg_messages.csv', index=False)